bigframes/functions/remote_function.py

```diff
@@ -285,38 +289,104 @@ def generate_cloud_function_main_code(self, def_, dir):
         #   ...
         # }
         # https://cloud.google.com/bigquery/docs/reference/standard-sql/remote-functions#input_format
-        code_template = textwrap.dedent(
-            """\
-        import cloudpickle
-        import functions_framework
-        from flask import jsonify
-        import json
-
-        # original udf code is in {udf_code_file}
-        # serialized udf code is in {udf_bytecode_file}
-        with open("{udf_bytecode_file}", "rb") as f:
-          udf = cloudpickle.load(f)
-
-        def {handler_func_name}(request):
-          try:
-            request_json = request.get_json(silent=True)
-            calls = request_json["calls"]
-            replies = []
-            for call in calls:
-              reply = udf(*call)
-              replies.append(reply)
-            return_json = json.dumps({{"replies" : replies}})
-            return return_json
-          except Exception as e:
-            return jsonify( {{ "errorMessage": str(e) }} ), 400
-        """
-        )
-
-        code = code_template.format(
-            udf_code_file=udf_code_file,
-            udf_bytecode_file=udf_bytecode_file,
-            handler_func_name=handler_func_name,
-        )
+        code = """\
+import cloudpickle
+import functions_framework
+from flask import jsonify
+import json
+"""
+        if is_row_processor:
+            code += """\
+import pandas as pd
+
+def get_pd_series(row):
+    row_json = json.loads(row)
+    col_names = row_json["names"]
+    col_types = row_json["types"]
+    col_values = row_json["values"]
+    index_names = row_json["index"]
+    dtype = row_json["dtype"]
+
+    # index and column names are not necessarily strings
+    # they are serialized as repr(repr(name)) at source
+    col_names = [eval(name) for name in col_names]
+    index_names = [eval(name) for name in index_names]
```

Comment:
Ok, I see now why `repr(repr(value))` is being used. eval makes me a bit uneasy, as it will execute arbitrary code. Als, repr/eval working nicely together isn't going to be guaranteed for more advanced types later on. For the primitive types we have right now, can we use `literal_eval` instead, or just use a non-eval deserialization entirely?

Reply:
great, literal_eval makes me feel better about this. One thing to note is that the eval-repr pattern only works for types that have literals in the python language. More complex/derived types like some datetime types, etc won't work with this pattern. I think it would be best to raise Errors on client side for these cases if posisble, until we can support serializing these types.

Reply:
added early type in .apply